conans/client/tools/files.py

```diff
@@ -383,3 +383,16 @@ def unix2dos(filepath):
 
 def dos2unix(filepath):
     _replace_with_separator(filepath, "\n")
+
+def rename(src, dst):
+    """
+    rename a file or folder to avoid "Access is denied" error on Windows
+    :param src: Path to the file or folder
+    :param dst: Destination file or folder
+    """
+    if platform.system() == "Windows":
+        retcode = os.system("robocopy /move /e /ndl /nfl".format(src, dst))
```

Comment:
Adding comments about the flags for future review:

```
/E :: copy subdirectories, including Empty ones.
/MOVE :: MOVE files AND dirs (delete from source after copying).
/NFL :: No File List - don't log file names.
/NDL :: No Directory List - don't log directory names.
```

Robocopy is installed by default in Windows even in those without professional license.

Reply:
I don't think it's available in all Windows versions. better add some check (`tools.which`)

Reply:
I checked that in a home edition version but maybe it is not there in a windows server?

Reply:
According to wikipedia: https://en.wikipedia.org/wiki/Robocopy , it is bundled with any version of Windows after XP.

Reply:
There are conan users stuck with Windows XP :(

Reply:
The latest version of this PR check for existence of robocopy.